wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt

```diff
@@ -116,26 +118,28 @@ class WireGenerator(
 }
 
 private fun parseFileDescriptor(fileDescriptor: DescriptorProtos.FileDescriptorProto, descs: DescriptorSource): ProtoFileElement {
-  val path = mutableListOf<Int>()
   val location = Location.get(fileDescriptor.name)
   val helper = SourceCodeHelper(fileDescriptor)
 
   val imports = mutableListOf<String>()
   val publicImports = mutableListOf<String>()
   val types = mutableListOf<TypeElement>()
 
+  val path = mutableListOf<Int>()
   path.clear()
   path.add(DescriptorProtos.FileDescriptorProto.MESSAGE_TYPE_FIELD_NUMBER)
   path.add(0) // placeholder for index
   for ((index, messageType) in fileDescriptor.messageTypeList.withIndex()) {
     path[1] = index
     types.add(parseMessage(path, helper, messageType, descs))
   }
-  // TODO: enums
-//  for (nestedType in fileDescriptor.enumTypeList) {
-//    types.add(parseEnum(nestedType, descs))
-//  }
 
+  val enumPaths = mutableListOf<Int>().apply { addAll(path) }
+  enumPaths.add(DescriptorProtos.DescriptorProto.ENUM_TYPE_FIELD_NUMBER)
```

Comment:
I had expected to just re-use `path` for each other kind of element. We don't want to keep appending to the path. The first two elements should be the enum field number and then the index. Also we need the field number from the _file_ descriptor, not the message descriptor
```suggestion
  path.clear()
  path.add(DescriptorProtos.FileDescriptorProto.ENUM_TYPE_FIELD_NUMBER)
  path.add(0) // placeholder for index
```

Although we could instead be clever and re-use existing indices. I just worry it might be error-prone -- like if code above does something to change length of `path`, then it would be wrong.
```kotlin
  path[0] = DescriptorProtos.FileDescriptorProto.ENUM_TYPE_FIELD_NUMBER
```

Reply:
The other idea is to model this path functionality with a new object but I was too afraid to do this while I don't quite understand how this should work haha

Reply:
Yes, that would probably be better. In my head, I'm trying to translate how I'd do this in Go (with a Slice) into something for the JVM. But lists in JVM != slices. Something like a persistent list would be closer...

Anyhow, let's merge as is, and I can push a change to clean this up and not try to re-use `path`.